code/modules/economy/cash.dm

```diff
@@ -75,14 +75,21 @@
 
 /obj/item/weapon/spacecash/bundle/attack_self()
 	var/amount = input(usr, "How many credits do you want to take? (0 to [src.worth])", "Take Money", 20) as num
+
+	if(QDELETED(src))
+		return 0
+
+	if(get_turf(src) != get_turf(usr))
```

Comment:
Use of `usr` in a non-interfacing proc. Should be refactored over to use the first argument of `attack_self`.

Also, consider using `if (!use_check(user))` here instead. It checks for adjacency by default, and also for a whole lot of other shit.